spec/predicates/test-results.md

```diff
@@ -0,0 +1,172 @@
+
+Authors:
+
+## Purpose
+
+This predicate type defines a generic schema to express the result of running
+tests in software supply chains. The schema may be extended to support different
+types of testing or specific test harnesses.
+
+## Use Cases
+
+Software development processes include several types of tests. This attestation
+can be used to express the results of running those tests. It can be used to
+verify:
+
+1.  that all tests were in fact run, and
+2.  that all required tests passed
+
+Therefore, each attestation corresponds to one invocation of a test suite, and
+may include the results of several individual tests.
+
+### Asserting Test Configurations Used
+
+The supply chain owner creates a policy that records the expected test
+configurations. During verification, the policy checks that the test attestation
+used the right configurations. A custom inspection may optionally parse the
+`testRun.link` field to verify attestation matches the test run.
+
+### Asserting Test Results
+
+In addition to the previous use case, the supply chain owner creates a policy
+verifying that test results passed. In the simplest case, the policy applies to
+all tests. Therefore, it asserts the contents of `result` and that
+`testRun.failedTests` is empty. In more nuanced cases, a subset of tests may
+matter. For example, the tested artifact may be an OS image that's to be
+deployed to three types of devices. As such, the test harness tests the new
+image on an instance of each device. When verifying the attestation prior to an
+image being installed on a device, it only matters that the tests passed on the
+corresponding test device and not necessarily the others.
+
+As before, a custom inspection may examine the `testRun.link` contents to verify
+the contents of the attestation.
+
+## Prerequisites
+
+Understanding of the
+[in-toto attestation specification](https://github.com/in-toto/attestation).
+
+## Model
+
+This predicate type includes one compulsory field, `result`, that describes the
+result of the test run. The `testRun` object can be used to communicate a link
+to the test run and list tests that passed, failed, and passed with warnings. It
+also may optionally contain a reference to the configuration for the test run.
+The expected `subject` for this are the source artifacts themselves or some
+reference to them such as a Git commit.
+
+## Schema
+
+```json
+{
+    "_type": "https://in-toto.io/Statement/v1",
+    "subject": [{...}],
+    "predicateType": "https://in-toto.io/attestation/test-result/v0.1",
+    "predicate": {
+        "result": "pass|fail",
+        "testRun": {
+            "link": "<URL>",
+            "configuration": ["<ResourceDescriptor>", ...],
+            "passedTests": ["<TEST_NAME>", ...],
+            "warnedTests": ["<TEST_NAME>", ...],
+            "failedTests": ["<TEST_NAME>", ...]
+        }
+    }
+}
+```
+
+### Parsing Rules
+
+This predicate follows the in-toto attestation parsing rules. Summary:
```

Comment:
Broader question, do we want every predicate to include a summarised copy of the framework's parsing rules? Or just to note where the predicate's parsing rules differ from the framework's parsing rules?

WDYT @in-toto/attestation-maintainers ? We can move this to a separate issue, the discussion should not block this PR.

Reply:
Speaking with my author of predicates hat on:

It _was_ a bit of work to go back and update each of them as the attestation framework wide rules changed. I could propose a change in ITE-9 that this is only to be used to define deviations from the regular rules.

Reply:
Yeah, maintainability seems much easier with this in one place (that we link to from the predicate?)

Reply:
Updated this one to only link to the spec.